### PR TITLE
feat: Filter Responses by hidden field values

### PR DESCRIPTION
**Pull request by gupta-piyush19**
## What does this PR do?
- Filter Responses by hidden field values
- Adds response filtering on hidden fields

Fixes:
https://github.com/formbricks/internal/issues/171

## How should this be tested?
- Create a survey with hidden fields in it
- Fill out the survey with hidden fields values
- filter the responses on the summary/response page

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/lib/surveys/surveys.ts [hunk 1]
```diff
@@ -189,8 +211,11 @@ export const getFormattedFilters = (
   dateRange: DateRange
 ): TResponseFilterCriteria => {
   const filters: TResponseFilterCriteria = {};
-  const [questions, tags, attributes, others, meta] = selectedFilter.filter.reduce(
-    (result: [FilterValue[], FilterValue[], FilterValue[], FilterValue[], FilterValue[]], filter) => {
+  const [questions, tags, attributes, others, meta, hiddenFields] = selectedFilter.filter.reduce(
```
Comment (pandeymangg) [inline apps/web/app/lib/surveys/surveys.ts:214]: pls use a `map` here instead of a reduce, it fits the use case better 😊

**Review by pandeymangg — CHANGES_REQUESTED** [review]
Thanks for the PR @gupta-piyush19 looks great 🚀 
just mentioned two small things, pls have a look 😊

**Review by pandeymangg — APPROVED** [review]
LGTM! 🎸

**Review by mattinannt — APPROVED** [review]
@gupta-piyush19 thanks for making the changes :-) Looks great! 💪

**Pull request merged**